src/record_syscall.cc

```diff
@@ -3748,6 +3749,10 @@ static void process_mmap(RecordTask* t, size_t length, int prot, int flags,
       TraceWriter::RECORD_IN_TRACE) {
     off64_t end = (off64_t)st.st_size - km.file_offset_bytes();
     t->record_remote(addr, min(end, (off64_t)km.size()));
+
+    if ((flags & MAP_SHARED) && !t->fd_table()->is_monitoring(fd)) {
```

Comment:
I think we should check whether the monitor is already a MmappedFileMonitor. If it's another kind of monitor we have a problem!